site/src/components/ErrorSummary/ErrorSummary.tsx

```diff
@@ -1,32 +1,112 @@
 import Button from "@material-ui/core/Button"
+import Collapse from "@material-ui/core/Collapse"
+import IconButton from "@material-ui/core/IconButton"
+import Link from "@material-ui/core/Link"
+import { makeStyles, Theme } from "@material-ui/core/styles"
+import CloseIcon from "@material-ui/icons/Close"
 import RefreshIcon from "@material-ui/icons/Refresh"
-import { FC } from "react"
+import { ApiError, getErrorDetail, getErrorMessage } from "../../api/errors"
+import { FC, useState } from "react"
 import { Stack } from "../Stack/Stack"
 
 const Language = {
   retryMessage: "Retry",
   unknownErrorMessage: "An unknown error has occurred",
+  moreDetails: "More",
+  lessDetails: "Less",
 }
 
 export interface ErrorSummaryProps {
-  error: Error | unknown
+  error: ApiError | Error | unknown
   retry?: () => void
+  dismissible?: boolean
+  defaultMessage?: string
 }
 
-export const ErrorSummary: FC<ErrorSummaryProps> = ({ error, retry }) => (
-  <Stack>
-    {!(error instanceof Error) ? (
-      <div>{Language.unknownErrorMessage}</div>
-    ) : (
-      <div>{error.toString()}</div>
-    )}
-
-    {retry && (
-      <div>
-        <Button onClick={retry} startIcon={<RefreshIcon />} variant="outlined">
-          {Language.retryMessage}
-        </Button>
-      </div>
-    )}
-  </Stack>
-)
+export const ErrorSummary: FC<ErrorSummaryProps> = ({
+  error,
+  retry,
+  dismissible,
+  defaultMessage,
+}) => {
+  const message = getErrorMessage(error, defaultMessage || Language.unknownErrorMessage)
+  const detail = getErrorDetail(error)
+  const [showDetails, setShowDetails] = useState(false)
+  const [isOpen, setOpen] = useState(true)
+
+  const styles = useStyles({ showDetails })
+
+  const toggleShowDetails = () => {
+    setShowDetails(!showDetails)
+  }
+
+  const closeError = () => {
+    setOpen(false)
+  }
+
+  if (!isOpen) {
+    return null
+  }
+
+  return (
+    <Stack>
+      <Stack className={styles.root}>
+        <Stack direction="row" alignItems="center" className={styles.message}>
+          <div>
+            <span className={styles.errorMessage}>{message}</span>
+            {!!detail && (
+              <Link onClick={toggleShowDetails} className={styles.detailsLink}>
```

Comment:
can we use an `aria-expanded` label here? https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded

Reply:
Great idea! Fixing this now.